include/exiv2/bmffimage.hpp

```diff
@@ -0,0 +1,177 @@
+// ***************************************************************** -*- C++ -*-
+/*
+ * Copyright (C) 2021 Exiv2 authors
+ * This program is part of the Exiv2 distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, 5th Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+// *****************************************************************************
+#include "exiv2lib_export.h"
+
+// included header files
+#include "image.hpp"
+
+// *****************************************************************************
+// namespace extensions
+namespace Exiv2
+{
+    EXIV2API bool enableBMFF(bool enable = true);
```

Comment:
Because the legal situation, we have two levels of protection from the bmff code.  First you have to build the library with buff support and secondly, you have to enable it at run-time.  In this way, existing applications (such a GIMP) will not be updated to support bmff even if the library/dll/shared-object is updated.  Applications have to request bmff support.

Reply:
Ah ok I see, thanks for the explanation!    

I still have no idea what the legal discussion is about and trusting you and the others that there isn't an actual problem :wink: